File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>

int get_length(long cc_number);

int main(void)
{
    long cc_number = get_long("CC Number: ");
    int length = get_length(cc_number);
    
    /* build array of numbers from cc_number */
    int cc_array[length];
    long num = cc_number;
    int index;
    for (int i = 0; i < length; i++)
    {
        index = length - i - 1;
        cc_array[index] = num % 10;
        num = num / 10;
    }
    
    /*get every other number, add & store in 2 variables, check if double digit*/
    int singles = 0;
    int doubles = 0;
    int d;
    num = cc_number;
    
    for (int i = 0; i < length; i ++)
    {
        index = length - i - 1;
        if (i % 2 == 0)/*add together*/
        {
            singles += cc_array[index]; 
        }
        else /*multiply by 2 and add together*/
        {
            d = cc_array[index] * 2;
            if (d > 9) /*add together digits of double-digit numbers*/
            {
                doubles += (d % 10) + (d / 10);
            }
            else
            {
                doubles += d;
            }
        }
    }
    /* check if last number of sum is 0 */
    int sum = singles + doubles;
    int is_zero = sum % 10;
    if (is_zero == 0)
    {
        if ((length == 15 && cc_array[0] == 3) && (cc_array[1] == 4 || cc_array[1] == 7))
        {
            printf("AMEX\n");
        } 
        else if ((length == 13 || length == 16) && (cc_array[0] == 4))
        {
            printf("VISA\n");  
        }
        else if (length == 16 && cc_array[0] == 5 && cc_array[1] > 0 && cc_array[1] < 6)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }    
    else
    {
        printf("INVALID\n");
    }
}
 
int get_length(long cc_number)
{
    int length = 0;
    do
    {
        length ++;
        cc_number = cc_number / 10;
    }
    while (cc_number > 0);
    return length;   
}

File: pset6/credit.py
from cs50 import get_int

# get cc number and length
cc_number = get_int("CC Number: ")
length = len(str(cc_number))

# build list of digits in cc number
cc_number_list = []
num = cc_number
for i in range(length):
    cc_number_list.insert(0, num % 10)
    num = int(num / 10)

#get every other number, add & store in 2 variables, check if double digit    
singles = 0
doubles = 0
num = cc_number
for i in range(length):
    index = length - i - 1
    if i % 2 == 0:
        singles += cc_number_list[index]
    else:
        d = cc_number_list[index] * 2
        if d > 9:
            doubles += (d % 10) + (int(d/10))
        else:
            doubles += d
#check if last number of sum is 0 & determine type of cc number
if (singles + doubles) % 10 == 0:
    if (length == 15 and cc_number_list[0] == 3) and (cc_number_list[1] == 4 or cc_number_list[1] == 7):
        print("AMEX")
    elif (length == 13 or length == 16) and cc_number_list[0] == 4:
        print("VISA")
    elif length == 16 and cc_number_list[0] == 5 and cc_number_list[1] > 0 and cc_number_list[1] < 6:
        print("MASTERCARD")
    else:
        print("INVALID")
else:
    print("INVALID")